packages/itwin/one-click-lca-widget/src/components/ExportModal.tsx

```diff
@@ -0,0 +1,358 @@
+/*---------------------------------------------------------------------------------------------
+* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+* See LICENSE.md in the project root for license terms and full copyright notice.
+*--------------------------------------------------------------------------------------------*/
+import "./ExportModal.scss";
+import { useCallback, useEffect, useMemo, useRef, useState } from "react";
+import { IModelApp } from "@itwin/core-frontend";
+import {
+  Alert,
+  Button,
+  LabeledInput,
+  Modal,
+  ProgressLinear,
+  ProgressRadial,
+  Text,
+} from "@itwin/itwinui-react";
+import {
+  SvgVisibilityHide,
+  SvgVisibilityShow,
+} from "@itwin/itwinui-icons-react";
+import {
+  JobCreation,
+  JobStatus,
+  Link,
+  OneClickLCAClient,
+} from "@itwin/insights-client";
+import logo from "../../public/logo/oneClickLCALogo.png";
+
+interface ExportProps {
+  isOpen: boolean;
+  close: () => void;
+  reportId: string | undefined;
+}
+
+interface OclcaTokenCache {
+  token: string;
+  exp: number;
+}
+
+const ExportModal = (props: ExportProps) => {
+  const MILI_SECONDS = 1000;
+  const PIN_INTERVAL = 1000;
+  const oneClickLCAClientApi = useMemo(() => new OneClickLCAClient(), []);
+
+  const [email, setEmail] = useState("");
+  const [password, setPassword] = useState("");
+  const [passwordIsVisible, showPassword] = useState(false);
+  const [signinError, showSigninError] = useState(false);
+  const [emailError, setEmailError] = useState(false);
+  const [cache, cacheToken] = useState<OclcaTokenCache>();
+
+  const [jobStatus, setJobStatus] = useState<JobStatus.StatusEnum>();
+  const [jobLink, setJobLink] = useState<Link>();
+
+  const isValidEmail = useCallback(() => {
+    return /\S+@\S+\.\S+/.test(email);
+  }, [email]);
+
+  const isValidPassword = useCallback(() => {
+    return password !== "";
+  }, [password]);
+
+  const validateSignin = useCallback(() => {
+    return cache?.token && cache?.exp > Date.now();
+  }, [cache]);
+
+  const [isSignedIn, setIsSignedIn] = useState(validateSignin());
+  const [isSigningIn, startSigningIn] = useState(false);
+  const intervalRef = useRef<NodeJS.Timer>();
```

Comment:
using type `NODEJS.Timer` is unsafe here, this will cause issues in the browser because the node api isnt available there. this will def cause issues in webpack 5 unless you add a polyfill, which is still a bad idea